### PR TITLE
Role and Permissions

### PR DESCRIPTION
**Pull request by dorozco94**

### CODE REVIEW

File: source/includes/_role_categories.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
+# Role Categories
+## List Role Categories
+
+```javascript
+fetch('https://core.eventtia.com/v1/role_categories/<id>', {
```
Comment (focampo92) [inline source/includes/_role_categories.md:5]: porque id en un list?

File: source/includes/_role_categories.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  "links": {
+      "first": "http://localhost:3000/api/v1/es/v1/role_categories?page[number]=1",
+      "last": "http://localhost:3000/api/v1/es/v1/role_categories?page[number]=1",
+      "prev": null,
+      "next": null
+    }
+}
+
+```
+This endpoint list role_categories return it
```
Comment (focampo92) [inline source/includes/_role_categories.md:38]: This endpoint allows you to list the role categories you have associated with your account.

File: source/includes/_role_categories.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+```http
+HTTP/1.1 404 Not Found
+{
+  "message": {
+    "error": "code: 128"
+  }
+}
+```
+
+This endpoint return an role category
```
Comment (focampo92) [inline source/includes/_role_categories.md:90]: This endpoint returns a role category given an id

File: source/includes/_role_categories.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+### HTTP Request
+
+`GET /v1/role_categories/:id`
+
+### Path Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+id | integer | id for the desired role_categories
```
Comment (focampo92) [inline source/includes/_role_categories.md:100]: id for the desired role category

File: source/includes/_role_categories.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+  }
+}
+```
+>Example of Unprocessable Entity (422) response:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+```
+
+This endpoint create an role category and return it
```
Comment (focampo92) [inline source/includes/_role_categories.md:146]: This endpoint creates a role category and returns it

File: source/includes/_role_categories.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+### HTTP Request
+
+`POST /v1/role_categories/`
+
+### Body Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+name | string | The Role category name.
```
Comment (focampo92) [inline source/includes/_role_categories.md:156]: Role category name

File: source/includes/_role_categories.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+```http
+HTTP/1.1 404 Not Found
+{
+  "message": {
+      "error": "code: 128"
+  }
+}
+```
+
+This endpoint update an role category and return it
```
Comment (focampo92) [inline source/includes/_role_categories.md:208]: This endpoint updates a role category and return it

File: source/includes/_role_categories.md [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+
+### HTTP Request
+
+`PUT /v1/role_categories/:id`
+
+### Path Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+id | integer | The id for the desired role_category
```
Comment (focampo92) [inline source/includes/_role_categories.md:218]: Id for the desired role category

File: source/includes/_role_categories.md [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+
+### Body Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+name | string | The role category name.
```
Comment (focampo92) [inline source/includes/_role_categories.md:224]: Role category name

File: source/includes/_role_categories.md [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+    "type": "role_categories",
+    "attributes": {
+        "id": 5,
+        "name": "grupo 10"
+      }
+  }
+}
+```
+
+This endpoint destroy a role category and return it
```
Comment (focampo92) [inline source/includes/_role_categories.md:256]: This endpoint destroys a role category and returns it

File: source/includes/_role_categories.md [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+
+### HTTP Request
+
+`DELETE /v1/role_categories/:id`
+
+### Path Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+id | integer | The id for the desired role category
```
Comment (focampo92) [inline source/includes/_role_categories.md:266]: Id for the desired role category

File: source/includes/_role_permissions.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
+# Role Permissions
+## List Role Permissions
+
+```javascript
+fetch('https://core.eventtia.com/v1/roles/<id>/role_permissions/<id>', {
```
Comment (focampo92) [inline source/includes/_role_permissions.md:5]: porque id en la URL ?

File: source/includes/_role_permissions.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+  "links": {
+      "first": "http://localhost:3000/api/v1/es/v1/roles/1/role_permissions?page[number]=1",
+      "last": "http://localhost:3000/api/v1/es/v1/roles/1/role_permissions?page[number]=1",
+      "prev": null,
+      "next": null
+    }
+}
+
+```
+This endpoint list role_permissions belongs to roles and permissions and return it
```
Comment (focampo92) [inline source/includes/_role_permissions.md:56]: This endpoint list all role permissions related to a given role

File: source/includes/_role_permissions.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+### HTTP Request
+
+`GET /v1/roles/:id/role_permissions/`
+
+### Path Parameters
+
+Parameter |  Type   | Description
+--------- | ------- | -----------
+id | integer | The id for the desired role
```
Comment (focampo92) [inline source/includes/_role_permissions.md:66]: Id for the desired role

File: source/includes/_roles.md [hunk 15]
```diff
@@ -0,0 +1,270 @@
...
+  "links": {
+    "first": "http://localhost:3000/api/v1/es/v1/roles?page[number]=1",
+    "last": "http://localhost:3000/api/v1/es/v1/roles?page[number]=1",
+    "prev": null,
+    "next": null
+  }
+}
+```
+
+This endpoint return a list of roles
```
Comment (focampo92) [inline source/includes/_roles.md:38]: This endpoint returns a list of roles that your account have configured

File: source/includes/_roles.md [hunk 16]
```diff
@@ -0,0 +1,270 @@
...
+```http
+HTTP/1.1 404 Not Found
+{
+  "message": {
+    "error": "code: 128"
+  }
+}
+```
+
+This endpoint return an role
```
Comment (focampo92) [inline source/includes/_roles.md:91]: This endpoint returns a role given an id

File: source/includes/_roles.md [hunk 17]
```diff
@@ -0,0 +1,270 @@
...
+  }
+}
+```
+>Example of Unprocessable Entity (422) response:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+```
+
+This endpoint create an role and return it
```
Comment (focampo92) [inline source/includes/_roles.md:146]: This endpoint creates a role and return it

File: source/includes/_roles.md [hunk 18]
```diff
@@ -0,0 +1,270 @@
...
+
+### HTTP Request
+
+`POST /v1/roles/`
+
+### Body Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+name | string | The Role's name.
```
Comment (focampo92) [inline source/includes/_roles.md:156]: Role's name

File: source/includes/_roles.md [hunk 19]
```diff
@@ -0,0 +1,270 @@
...
+    }
+}
+```
+>Example of Unprocessable Entity (422) response: 
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+```
+
+This endpoint update an role and return it
```
Comment (focampo92) [inline source/includes/_roles.md:211]: This endpoint allows you to update a role information and associate or revoke permissions to the users that have associated it.

File: source/includes/_roles.md [hunk 20]
```diff
@@ -0,0 +1,270 @@
...
+
+### HTTP Request
+
+`PUT /v1/roles/:id`
+
+### Path Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+id | integer | The id for the desired role
```
Comment (focampo92) [inline source/includes/_roles.md:221]: Id of the desired role

File: source/includes/_roles.md [hunk 21]
```diff
@@ -0,0 +1,270 @@
...
+
+### Body Parameters
+
+Parameter | Type | Description
+--------- | ---- | -----------
+name | string | The Role's name.
```
Comment (focampo92) [inline source/includes/_roles.md:227]: Role's name

File: source/includes/_roles.md [hunk 22]
```diff
@@ -0,0 +1,270 @@
...
+    "type": "roles",
+    "attributes": {
+        "id": 4,
+        "name": "registro"
+    }
+  }
+}
+```
+
+This endpoint destroy a role and return it
```
Comment (focampo92) [inline source/includes/_roles.md:260]: This endpoint destroys a role and return it

File: source/includes/_users.md [hunk 23]
```diff
@@ -17,21 +17,34 @@ fetch('https://core.eventtia.com/v1/users/', {
 ```http
 HTTP/1.1 200 OK
 {
-  "data": [
+"data": [
     {
-      "id": 1,
+      "id": "1",
```
Comment (focampo92) [inline source/includes/_users.md:22]: porque el id como string?

File: source/includes/_users.md [hunk 24]
```diff
@@ -175,10 +209,14 @@ This endpoint create an user and return it
 
 Parameter | Type | Description
 --------- | ---- | -----------
-name | string | The User's name.
+first_name | string | The User's first name.
+last_name | string | The User's last name.
 email | string | The User's corporate email.
 phone | string | The User's phone.
 password | string | The User's password.
+is_admin | bolean | The User's is_admin.
```
Comment (focampo92) [inline source/includes/_users.md:217]: specifies whether the user you are creating is an administrator or not

File: source/includes/_users.md [hunk 25]
```diff
@@ -260,6 +314,8 @@ name | string | The User's name.
...
```
Comment (focampo92) [inline source/includes/_users.md:317]: specifies whether the user you are creating is an administrator or not.

File: source/includes/_users.md [hunk 26]
```diff
@@ -260,6 +314,8 @@ name | string | The User's name.
...
+role_category_ids | array | The User's role category ids.
```
Comment (focampo92) [inline source/includes/_users.md:318]: Specifies the categories the user is associated with. This is required in case your user isn't an administrator.

**Review by focampo92 — CHANGES_REQUESTED** [review]
Hay ciertas cosas que debemos ajustar. 

Good Job!

File: source/includes/_role_categories.md (hunk 1 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: corrected

File: source/includes/_role_categories.md (hunk 2 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 3 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 4 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 5 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 6 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 8 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 9 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 10 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_categories.md (hunk 11 above)
Comment (dorozco94) [reply source/includes/_role_categories.md]: done

File: source/includes/_role_permissions.md (hunk 12 above)
Comment (dorozco94) [reply source/includes/_role_permissions.md]: done

File: source/includes/_role_permissions.md (hunk 13 above)
Comment (dorozco94) [reply source/includes/_role_permissions.md]: done

File: source/includes/_role_permissions.md (hunk 14 above)
Comment (dorozco94) [reply source/includes/_role_permissions.md]: done

File: source/includes/_roles.md (hunk 15 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 16 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 17 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 18 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 19 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 20 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 21 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_roles.md (hunk 22 above)
Comment (dorozco94) [reply source/includes/_roles.md]: done

File: source/includes/_users.md (hunk 23 above)
Comment (dorozco94) [reply source/includes/_users.md]: done

File: source/includes/_users.md (hunk 24 above)
Comment (dorozco94) [reply source/includes/_users.md]: done

File: source/includes/_users.md (hunk 25 above)
Comment (dorozco94) [reply source/includes/_users.md]: done

File: source/includes/_users.md (hunk 26 above)
Comment (dorozco94) [reply source/includes/_users.md]: done

**Review by focampo92 — APPROVED** [review]

**Review by PierreQuintero-eventtia — APPROVED** [review]

**Pull request merged**